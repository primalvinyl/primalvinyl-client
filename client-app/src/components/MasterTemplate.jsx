import React from 'react';
import Header from './Header';
import Footer from './Footer';
import './MasterTemplate.scss';

function MasterTemplate({ children }) {
    return (
        <div className="master-template">
            <div>
                <Header />
                <main>
                    {children}
                </main>
            </div>
            <Footer />
        </div>
    );
}

export default MasterTemplate;
