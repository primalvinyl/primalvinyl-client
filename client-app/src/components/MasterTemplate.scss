@import './styles/partials/breakpoints';
@import './styles/partials/variables';

.master-template {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    main {
        padding: 1em 0;
    }

    @include breakpoint(lg-down) {
        main {
            padding: 1em;
        }
    }
}
