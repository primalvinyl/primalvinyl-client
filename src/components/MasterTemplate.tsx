import React from 'react';
import ErrorBoundary from './ErrorBoundary';
import Header from './Header';
import Footer from './Footer';
import './MasterTemplate.scss';

const MasterTemplate = ({ children }: MasterTemplateProps): React.ReactElement => {
    return (
        <div className="master-template">
            <div>
                <Header />
                <main>
                    <ErrorBoundary>
                        {children}
                    </ErrorBoundary>
                </main>
            </div>
            <Footer />
        </div>
    );
}

type MasterTemplateProps = {
    children: React.ReactElement;
};

export default MasterTemplate;
