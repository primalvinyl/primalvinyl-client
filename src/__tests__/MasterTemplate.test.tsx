import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MasterTemplate from '../components/MasterTemplate';

describe('MasterTemplate', () => {
    it('renders', () => {
        const myWrapper = shallow(
            <MemoryRouter>
                <MasterTemplate>Test</MasterTemplate>
            </MemoryRouter>
        );
        expect(myWrapper.exists());
    });
    it('matches snapshot', () => {
        const component = Renderer.create(
            <MemoryRouter>
                <MasterTemplate>Test</MasterTemplate>
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
