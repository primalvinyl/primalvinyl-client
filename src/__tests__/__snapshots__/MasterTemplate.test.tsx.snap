// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterTemplate matches snapshot 1`] = `
<div
  className="root"
>
  <div>
    <header
      className="root"
    >
      <div
        className="wrapper"
      >
        <img
          alt=""
          src="monster.png"
        />
        <a
          className="logotype"
          href="/"
          onClick={[Function]}
        >
          <h1>
            Primal Vinyl
          </h1>
        </a>
      </div>
    </header>
    <main>
      <div>
        Test
      </div>
    </main>
  </div>
  <footer
    className="root"
  >
    <div
      className="wrapper"
    >
      <div
        className="logotype"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <h2>
            Primal Vinyl
          </h2>
        </a>
        <div
          className="copyright"
        >
          Â© 
          2021
        </div>
      </div>
    </div>
  </footer>
</div>
`;
